src/engine/yaml.rs

```diff
@@ -83,4 +90,29 @@ three: baz
         let result: ParsedEntityStruct<FrontMatter> = matter.parse_with_struct(input).unwrap();
         assert_eq!(result.data, data_expected);
     }
+
+    #[test]
+    fn non_string_keys() {
+        let matter: Matter<YAML> = Matter::new();
+        let input = r#"---
+1: foo
+true: bar
+three: baz
+---"#;
+        #[derive(Deserialize, PartialEq, Debug)]
+        struct FrontMatter {
```

Comment:
Please add the null key in the test too.

Reply:
Done, thanks :)